notebooks/final/_04-tb-testing-your-causal-graph.md

```diff
@@ -0,0 +1,455 @@
+This implies that shuffling / permuting $X_2$ should make no difference for predicting $X_1$ once we've included $Z$ while predicting.
+
+In other words, one's ability predict to predict $X_1$ should not depend on whether one uses the original $X_2$ or the permuted $X_2$, as long as one conditions on $Z$ when predicting $X_1$.
+In this notebook, we test this invariance using a simple predictive model, linear regression, but in general, one need not and should not restrict oneself to linear predictive models. Upon estimating each regression, we compare $r^2$ as a measure of predictive ability when using $Z$ and the original $X_2$ versus $r^2$ when using $Z$ and the permuted $X_2$.
+
+For this test, we'll use the following conditional independence assumption implied by the causal graph above:<br>
+$
+P \left( \textrm{Travel Time} \mid \textrm{Travel Cost}, \textrm{Travel Distance} \right) = P \left( \textrm{Travel Time} \mid \textrm{Travel Distance} \right)
+$
+
+```python
+time_array = drive_alone_df[TIME_COLUMN].values
+cost_array = drive_alone_df[COST_COLUMN].values
+distance_array = drive_alone_df[DISTANCE_COLUMN].values
+
+oi.visual_permutation_test(
+    time_array,
+    cost_array,
+    z_array=distance_array,
+    num_permutations=NUM_PERMUTATIONS,
+    permutation_color=PERMUTED_COLOR)
+```
+
+### Caveats and pitfalls
+When testing conditional mean independence (i.e., $E \left[ X_1 \mid X_2, Z \right] = E \left[ X_1 \mid Z \right]$), there are four potential issues of interest.
+
+The first issue is misspecification of $E \left[ X_1 \mid Z \right]$.
+If the conditional mean of $X_1$ depends on un-modeled functions of $Z$ and $Z \rightarrow X_2$, then the inclusion of $X_2$ in one's model may serve as a proxy for the un-modeled function of $Z$.
+Such proxy behavior would lead one to observe "inflated" values of $r^2$ when modeling $E \left[ X_1 \mid X_2, Z \right]$, thus increasing the probability that one will falsely reject the null-hypothesis of conditional independence.
+
+In other words, to guard against higher-than-nominal probabilities of type-1 error, one needs to guard against underfitting of the model for $E \left[ X_1 \mid Z \right]$ **before** computing one's test-statistic and (permutation-based) reference-distribution.
+
+The second issue is misspecification of $E \left[ X_1 \mid X_2, Z \right]$.
+If one's model for $E \left[ X_1 \mid X_2, Z \right]$ is underfit with respect to $X_2$, then one's test-statistic ($r^2$) will be lower than it should be under accurate specification.
+This leads to lower probability of correctly rejecting the null-hypothesis.
+
+The last two issues are the converse of the problems and effects above.
+If one's model for $E \left[ X_1 \mid Z \right]$ is overfit, then the reference-distribution of the test statistic will be artificially high.
+This will lead to lower probability of correctly rejecting the null hypothesis of conditional independence.
```

Comment:
I suggest we rephrase this to standard hypothesis testing conventions. Instead of "lower the probability of correctly rejecting the null hypothesis", I would use "this will increase the probability of failing to reject the null hypothesis when it's true".

Reply:
Just addressed (although not using the exact wording you suggested).